back-end/java/src/main/java/controller/BayController.java

```diff
@@ -22,6 +22,24 @@
     };
 
     public static Handler addBay = ctx ->{
+
+        //Authentication
+        String loginEmail = ctx.formParam("loginEmail");
+        if (loginEmail==null){
+            ctx.json(new Status("No `loginEmail` Provided (The email of the user making the changes)"));
+            return;
+        }
+        String loginPassword = ctx.formParam("loginPassword");
+        if (loginPassword==null){
+            ctx.json(new Status("No `loginPassword` Provided (The password of the user making the changes)"));
+            return;
+        }
+        User user = UserDAO.checkLogin(loginEmail, loginPassword);
+        if (user==null){
```

Comment:
Should you also check the user's account type?

Reply:
Made those changes